hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/KeyOutputStream.java

```diff
@@ -484,6 +499,9 @@ public void close() throws IOException {
     closed = true;
     try {
       handleFlushOrClose(StreamAction.CLOSE);
+      if (!isException) {
+        Preconditions.checkArgument(writeOffset == offset);
```

Comment:
At this point in the code I am not the fan of throwing an IllegalArgumentException, wouldn't it be better to keep this as an IOException with a specific explaining message?

Reply:
I would prefer to have the precondition check intact, as its just in place to validate the logic. If its not met, the code is buggy. Also, throwing Runtime exception is okay as this exception does not need be caught and handled anywhere in general.